### PR TITLE
Dev dns microsoft.network 2018 05 01

### PR DESCRIPTION
**Pull request by vladrai**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by dsgouda** [comment]
Perfect, thanks @vladrai taking a look

File: specification/dns/resource-manager/Microsoft.Network/stable/2018-05-01/dns.json [hunk 1]
```diff
@@ -1021,6 +1072,43 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/getDnsResourceReference": {
+      "post": {
```
Comment (dsgouda) [inline specification/dns/resource-manager/Microsoft.Network/stable/2018-05-01/dns.json]: Is there a reason this is a `POST` operation, looks more like a `GET` operation

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great for the most part

**Comment by dsgouda** [comment]
Please address CI issues if any are reported

**Comment by dsgouda** [comment]
@vladrai please address the linter errors reported

File: specification/dns/resource-manager/Microsoft.Network/stable/2018-05-01/dns.json (hunk 1 above)
Comment (vladrai) [reply specification/dns/resource-manager/Microsoft.Network/stable/2018-05-01/dns.json]: It requires a list of resource Ids in the body. We already reviewed all API changes with ARM team.

**Review by dsgouda — APPROVED** [review]
LGTM

**Review by dsgouda — APPROVED** [review]
LGTM

File: specification/dns/resource-manager/readme.md [hunk 2]
```diff
@@ -4,31 +4,59 @@
...
+```yaml $(tag) == 'package-2018-05'
+input-file:
+  - Microsoft.Network/stable/2018-05-01/dns.json
+directive:
+  - where:
+      - $.paths
+    suppress:
+      - OperationsAPIImplementation
+  - where:
+      - $.definitions.RecordSetProperties.properties.TTL
```
[the remark this answers is not in the document]
Comment (dsgouda) [reply specification/dns/resource-manager/readme.md]: Please undo these changes

[the remark this answers is not in the document]
Comment (karravi-msft) [reply specification/dns/resource-manager/readme.md]: Deepak.. the same suppression is there for the previous api versions as well.. thats why we tried to suppress them

[the remark this answers is not in the document]
Comment (dsgouda) [reply specification/dns/resource-manager/readme.md]: Thanks for confirming

**Review by dsgouda — APPROVED** [review]
LGTM subject to CI failures

**Pull request merged**